qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/UpgradeSteps.java

```diff
@@ -29,7 +38,15 @@ public void finish() {
 		complete();
 	}
 
-	@Given("User enters credentials")
+	@Given("User is on login page")
+	public void visitLoginPage() {	
+		assertTrue(textExists(HEADER_TEXT));
+		assertTrue(textExists(SUPER_USER));
+		assertTrue(textExists(OPENMRS_VERSION));
```

Comment:
Will this test pass when the openmrs version is 3.x?

Reply:
> Will this test pass when the openmrs version is 3.x?

no,

however, one challenge I got in testing out the version is that OpenMRS version evolves over time so we are currently in 2.x version and  when the platform version reaches 3.x then the test will be adjusted accordingly. 

@dkayiwa do you have a suggestion on the subject?

Reply:
@kdaud Why do you increase our maintenance burden?

Reply:
> @kdaud Why do you increase our maintenance burden?

@dkayiwa your question is too smart. Well, increasing the maintenance burden is not a good idea.

Reply:
I guess you can now come up with a smart solution! 😊